src/commands/utils/birthday/functions/listBirthday.ts

```diff
@@ -8,42 +8,49 @@ const birthdayListEmbed = (fields: Array<any>) => makeEmbed({
 });
 
 export async function handleListBirthday(interaction: ChatInputCommandInteraction<'cached'>) {
-    const birthdays = await Birthday.find({}).sort({ day: 1 }); // Only day sort required, months are bucketized
-    const members = await interaction.guild!.members.fetch();
-
-    const monthBuckets: Array<string | Array<any>> = [
-        ['January', []],
-        ['February', []],
-        ['March', []],
-        ['April', []],
-        ['May', []],
-        ['June', []],
-        ['July', []],
-        ['August', []],
-        ['September', []],
-        ['October', []],
-        ['November', []],
-        ['December', []],
-    ];
-
-    for (const birthday of birthdays) {
-        const member = members.get(birthday.userID!);
-
-        if (member) {
-            monthBuckets[birthday.utcDatetime!.getUTCMonth()][1].push(`${member.displayName} - ${birthday.day}/${birthday.month} (Z${birthday.timezone! < 0 ? '' : '+'}${birthday.timezone})`);
+    await interaction.deferReply();
+
+    try {
+        const birthdays = await Birthday.find({}).sort({ day: 1 }); // Only day sort required, months are bucketized
+        const members = await interaction.guild!.members.fetch();
+
+        const monthBuckets: Array<string | Array<any>> = [
+            ['January', []],
+            ['February', []],
+            ['March', []],
+            ['April', []],
+            ['May', []],
+            ['June', []],
+            ['July', []],
+            ['August', []],
+            ['September', []],
+            ['October', []],
+            ['November', []],
+            ['December', []],
+        ];
+
+        for (const birthday of birthdays) {
+            const member = members.get(birthday.userID!);
+
+            if (member) {
+                monthBuckets[birthday.utcDatetime!.getUTCMonth()][1].push(`${member.displayName} - ${birthday.day}/${birthday.month} (Z${birthday.timezone! < 0 ? '' : '+'}${birthday.timezone})`);
```

Comment:
Did you test that if a birthday has no utcDateTime? Does it then just add another array in the month buckets (which it can't due to it being const) or fail in some way?

Reply:
Birthday will always have a utcDateTime. It is populated in the setBrithday command. To be fair, these birthday commands could do with an update as they're just a port from the old bot.

Perhaps an update can be done in a future PR

Reply:
Just asking because of the !, since that's implying that there could be no utcDateTime.just don't know how the array is behaving with that, but since it's just copied from the old bot, it's fine with me

Reply:
Yes, if I remember the linting thought it could not exist. But, its ensured to exist in the other command therefore it does always exist. But in this file it doesn't know that